kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2017 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.querywithcqlfilters
+
+import android.graphics.Color
+import android.os.Bundle
+import android.util.Log
+import android.widget.DatePicker
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SwitchCompat
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.data.OgcFeatureCollectionTable
+import com.esri.arcgisruntime.data.QueryParameters
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.GeometryEngine
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.TimeExtent
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.querywithcqlfilters.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.sample.querywithcqlfilters.databinding.CqlFiltersLayoutBinding
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import java.util.*
+
+
+class MainActivity : AppCompatActivity() {
+
+    // Keep loadable in scope to avoid garbage collection
+    private lateinit var ogcFeatureCollectionTable: OgcFeatureCollectionTable
+
+    // List of CQL where clauses
+    private val cqlQueryList: List<String> = listOf(
+        "F_CODE = 'AP010'",
+        "{ \"eq\" : [ { \"property\" : \"F_CODE\" }, \"AP010\" ] }",
+        "F_CODE LIKE 'AQ%'",
+        "{\"and\":[{\"eq\":[{ \"property\" : \"F_CODE\" }, \"AP010\"]},{ \"before\":" +
+                "[{ \"property\" : \"ZI001_SDV\"},\"2013-01-01\"]}]}",
+        ""
+    )
+
+    // Current selected where query
+    private var cqlQueryListPosition = 4
+
+    // Number of features query should return. Default is 1000
```

Comment:
I read this as the default value the API sets is 1000, which isn't the case (I think Tanner said it was 10!): we could go with the following comment which is used in the design for the other OGC API samples and provides some helpful context to the user:

      // set a limit of 1000 on the number of returned features per request, the default on some services could be as low as 10

Reply:
That's a good point, I've added it in!